arangod/Aql/Ast.cpp

```diff
@@ -1912,7 +1912,7 @@ void Ast::validateAndOptimize(transaction::Methods& trx) {
     std::unordered_set<std::string> writeCollectionsSeen;
     std::unordered_map<std::string, int64_t> collectionsFirstSeen;
     std::unordered_map<Variable const*, AstNode const*> variableDefinitions;
-    RegexCache regexCache;
+    AqlFunctionsInternalCache regexCache;
```

Comment:
the variable name should be changed, because `regexCache` is not just a cache for regexes anymore.